tests/admin/views/test_releases_v2.py

```diff
@@ -1474,3 +1631,245 @@ def test_post_add_scheduled_change_base_and_locale(api, firefox_56_0_build1):
+    assert ret.status_code == 403, ret.data
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_delete_release(api):
+    ret = api.delete("/v2/releases/Firefox-65.0-build1")
+    assert ret.status_code == 200, ret.data
+
+    assert len(dbo.releases_json.t.select(dbo.releases_json.name).where(dbo.releases_json.name == "Firefox-65.0-build1").execute().fetchall()) == 0
+    assert len(dbo.release_assets.t.select(dbo.release_assets.name).where(dbo.release_assets.name == "Firefox-65.0-build1").execute().fetchall()) == 0
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_delete_scheduled_insert(api):
+    ret = api.delete("/v2/releases/Firefox-64.0-build1")
+    assert ret.status_code == 200, ret.data
+
+    assert (
+        len(dbo.releases_json.scheduled_changes.t.select(dbo.releases_json.name).where(dbo.releases_json.name == "Firefox-64.0-build1").execute().fetchall())
+        == 0
+    )
+    assert (
+        len(dbo.release_assets.scheduled_changes.t.select(dbo.release_assets.name).where(dbo.release_assets.name == "Firefox-64.0-build1").execute().fetchall())
+        == 0
+    )
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_delete_nonsplit_release(api):
+    ret = api.delete("/v2/releases/CDM-16")
+    assert ret.status_code == 200, ret.data
+
+    assert len(dbo.releases_json.t.select(dbo.releases_json.name).where(dbo.releases_json.name == "CDM-16").execute().fetchall()) == 0
+    assert len(dbo.release_assets.t.select(dbo.release_assets.name).where(dbo.release_assets.name == "CDM-16").execute().fetchall()) == 0
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_delete_fails_for_readonly_release(api):
+    dbo.releases_json.t.update(values={"read_only": True}).where(dbo.releases_json.name == "Firefox-65.0-build1").execute()
+
+    ret = api.delete("/v2/releases/Firefox-65.0-build1")
+    assert ret.status_code == 400, ret.data
+    assert "Cannot delete" in ret.json["exception"]
+
+
+@pytest.mark.usefixtures("releases_db", "mock_verified_userinfo")
+def test_set_readonly_404(api):
+    ret = api.put("/v2/releases/Firefox-22.0-build1/read_only", json={"read_only": True, "old_data_version": 1})
+    assert ret.status_code == 404, ret.data
+
+
+@pytest.mark.usefixtures("releases_db")
+def test_set_readonly_without_permission(api, mock_verified_userinfo):
+    mock_verified_userinfo("notbob")
+    ret = api.put("/v2/releases/Firefox-60.0b3-build1/read_only", json={"read_only": True, "old_data_version": 1})
+    assert ret.status_code == 403, ret.data
+
+
+@pytest.mark.usefixtures("releases_db")
+def test_set_readwrite_without_permission(api, mock_verified_userinfo):
+    mock_verified_userinfo("notbob")
+    dbo.releases_json.t.update(values={"read_only": True}).where(dbo.releases_json.name == "CDM-16").execute()
+    ret = api.put("/v2/releases/CDM-16/read_only", json={"read_only": True, "old_data_version": 1})
```

Comment:
Nit, I think you wanted `"read_only": False` here, even though the permission check fails prior to using it.